internal/textwrap/textwrap_test.go

```diff
@@ -0,0 +1,213 @@
+package textwrap
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestDedent(t *testing.T) {
+	tests := []struct {
+		name     string
+		input    string
+		expected string
+	}{
+		{
+			name: "Basic indentation",
+			input: `
+\t\t\tHello
+\t\t\t\tWorld
+\t\t\t  Foo
+\t\t\t    Bar
+\t\t\t`,
+			expected: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+		},
+		{
+			name: "No common indentation",
+			input: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+			expected: `
+Hello
+\tWorld
+  Foo
+    Bar
+`,
+		},
+		{
+			name: "All lines indented",
+			input: `
+\t\t\tHello
+\t\t\tWorld
+\t\t\tFoo
+\t\t\tBar
+\t\t\t`,
+			expected: `
+Hello
+World
+Foo
+Bar
+`,
+		},
+		{
+			name: "Mixed tabs and spaces",
```

Comment:
I think I might like to see a test like (and maybe I missed this)
```
\tHello
 World
 ```
 
 to define if tabs and spaces are treated equivalently.

Reply:
Yeah I think this case should have a bare 3 tab and 3 tab+1 space lines for completeness.

Reply:
Done

Reply:
I'm not sure this covers what I had in mind. This still just strips 3x \t. I was curious if \t and ' ' are treated equivalent and this, from what I can see, may not answer that question?

like does 
```
\tHello
 World
 ```
 dedent to
 ```
 Hello
 World
 ```
 or remain
 ```
 \tHello
  World
  ```

Reply:
They're not equivalent so they'd remain not indented. I'll add a test for this.

Reply:
Done